### PR TITLE
Enforce packages list name

### LINKED ISSUE #8330: debian-12-minimal packages has unwanted extraneous packages installed
### Qubes OS release

R4.2

### Brief summary

Unwanted extraneous packages installed by default in the `debian-12-minimal` template.

### Steps to reproduce

    sudo qvm-template --enablerepo=qubes-templates-itl-testing install debian-12-minimal

Start Template.

    dpkg -l | grep firefox

### Expected behavior

Packages such as firefox not being installed. Also keepassxc, network-manager-gnome, thunderbird are installed.

### Actual behavior

Packages such as firefox are installed.

### Additional information

Also potentially other unwanted extraneous packages might be installed.

* affects Whonix: yes, likely.
* effect on Whonix: Since the Whonix template is based on Debian minimal template, these packages are also installed in Whonix-Gateway, which is unwanted.

This is a regression.

It is likely unintentional because file https://github.com/QubesOS/qubes-builder-debian/blob/main/template_debian/packages_bullseye_minimal.list wasn't modified.

The bug might have been introduced by qubes-builder version 1 to version 2 migration?

### PR DESCRIPTION
**Pull request by fepitre**
Fixes https://github.com/QubesOS/qubes-issues/issues/8330

### CODE REVIEW

File: template_debian/02_install_groups.sh [hunk 1]
```diff
@@ -59,10 +59,17 @@ info ' Distribution specific steps (install systemd, add sources, etc)'
 buildStep "$0" "${DIST_CODENAME}"
 
 #### '----------------------------------------------------------------------
-info " Installing extra packages in script_${DIST_CODENAME}/packages.list file"
+info " Installing extra packages from packages list"
 #### '----------------------------------------------------------------------
+
+packages_list_basename=packages_${DIST_CODENAME}
+if [ -z "${TEMPLATE_FLAVOR}" ]; then
```
Comment (marmarek) [inline template_debian/02_install_groups.sh]: -n ?

**Comment by marmarek** [comment]
This might fix debian-12-minimal template, but I don't think it will help with whonix templates, because they only use "minimal" option, not flavor.
Maybe "minimal" option needs special handling, as it's already the case at https://github.com/QubesOS/qubes-builder-debian/blob/main/template_debian/02_install_groups.sh#L45 ?

File: template_debian/02_install_groups.sh (hunk 1 above)
Comment (fepitre) [reply template_debian/02_install_groups.sh]: yes sorry

**Comment by fepitre** [comment]
> This might fix debian-12-minimal template, but I don't think it will help with whonix templates, because they only use "minimal" option, not flavor. Maybe "minimal" option needs special handling, as it's already the case at https://github.com/QubesOS/qubes-builder-debian/blob/main/template_debian/02_install_groups.sh#L45 ?

I think this is the original problem for the lookup function for packages list resolves multiples files. I'll find a solution.

**Comment by marmarek** [comment]
Building minimal template with legacy builder (for R4.1) doesn't work for me, or more specifically, appears to build the full template instead.

**Comment by marmarek** [comment]
Bad news, this time builderv2 minimal build includes non-minimal `packages_qubes.list` :/

**Comment by fepitre** [comment]
According to our discussion, it was correctly using the right `packages_qubes_minimal.list` file but it was missing `no-recommends` in https://github.com/QubesOS/qubes-continuous-integration/commit/b52838079b4d7cf5a48a80c116b2245df58fea1f.

**Review by marmarek — APPROVED** [review]

**Pull request merged**